File: codex_poting/java/src/main/java/com/example/demo/mybatis/entity/BookWithPublisherName.java
package com.example.demo.mybatis.entity;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class BookWithPublisherName {
    private Long id;
    private String title;
    private String author;
    private LocalDate releaseDate;
    private Long publisherId;
    private String publisherName;
    private Long genreId;
    private String genreName;
    private String isbn;
    private Integer salesUnitPrice;
    private LocalDateTime updateAt;
    private Long version;
    private List<BookStockWithStoreName> bookStockList = new ArrayList<>();
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/mybatis/entity/BookWithPublisherName.kt
package com.example.demo.mybatis.entity

import java.time.LocalDate
import java.time.LocalDateTime

class BookWithPublisherName {
    var id: Long? = null
    var title: String? = null
    var author: String? = null
    var releaseDate: LocalDate? = null
    var publisherId: Long? = null
    var publisherName: String? = null
    var genreId: Long? = null
    var genreName: String? = null
    var isbn: String? = null
    var salesUnitPrice: Int? = null
    var updateAt: LocalDateTime? = null
    var version: Long? = null
    var bookStockList: MutableList<BookStockWithStoreName?>? = ArrayList<BookStockWithStoreName?>()
}
